ai-services/internal/pkg/runtime/podman/podman.go

```diff
@@ -122,7 +122,18 @@ func (pc *PodmanClient) ListContainers(filters map[string][]string) (any, error)
 }
 
 func (pc *PodmanClient) StopPod(id string) error {
-	_, err := pods.Stop(pc.Context, id, &pods.StopOptions{})
+	inpectReport, err := pc.InspectPod(id)
+	if err != nil {
+		return fmt.Errorf("failed to inspect pod: %w", err)
+	}
+
+	for _, container := range inpectReport.Containers {
+		err := containers.Stop(pc.Context, container.ID, nil)
+		if err != nil {
+			return fmt.Errorf("failed to stop pod container %s; err: %w", container.ID, err)
+		}
+	}
+	_, err = pods.Stop(pc.Context, id, &pods.StopOptions{})
```

Comment:
aren't pods stop not stopping the undelying containers?

Reply:
it does, but not in correct order. This is fixed in v5 of podman, but for now we will have to go with this until we upgrade podman to v5
Ref: https://github.com/containers/podman/pull/25169

Reply:
Infra container should be stopped at last, I missed that. Let me modify the changes

Reply:
updated the PR, it skips the infra container and let others be stopped. 
and in the end, the infra is stopped from pods.Stop().

This should ideally be handled by podman, but code is getting in this part of sdk: https://github.com/containers/podman/blob/main/libpod/pod_api.go#L164


@mkumatag I looked into it, and seems like infra container is stopped first as podman is not able to build the dependency graph (what I think), thats why cgroup and other stale resources are there when we try to start the same pods again.

Open for suggestions